client/src/components/Profile/ContactsCard.tsx

```diff
@@ -1,242 +1,177 @@
-        value: epamEmail,
-        key: 'epamEmail',
-        rules: [{ type: 'email', message: 'Email is not valid' }],
-      },
-      {
-        name: 'E-mail',
-        value: email,
-        key: 'email',
-        rules: [{ type: 'email', message: 'Email is not valid' }],
-      },
-      {
-        name: 'Telegram',
-        value: telegram,
-        key: 'telegram',
-      },
-      {
-        name: 'Phone',
-        value: phone,
-        key: 'phone',
-      },
-      {
-        name: 'Skype',
-        value: skype,
-        key: 'skype',
-      },
-      {
-        name: 'Notes',
-        value: notes,
-        key: 'notes',
-      },
-      {
-        name: 'LinkedIn',
-        value: linkedIn,
-        key: 'linkedIn',
-      },
-    ];
-
-    const filledContacts = contacts.filter(({ value }: Contact) => value);
-
-    return (
-      <CommonCard
-        title="Contacts"
-        icon={<ContactsOutlined />}
-        content={
-          filledContacts.length ? (
-            <List
-              itemLayout="horizontal"
-              dataSource={filledContacts}
-              renderItem={({ name, value, key }: Contact) => (
-                <List.Item>
-                  <Text strong>{name}:</Text>{' '}
-                  {key !== 'linkedIn' ? (
-                    <>
-                      {value}
-                      {key === 'email' && !isDataPendingSave && (!connections.email || !connections.email.enabled) ? (
-                        <EmailConfirmation
-                          connection={connections.email}
-                          sendConfirmationEmail={sendConfirmationEmail}
-                        />
-                      ) : null}
-                    </>
-                  ) : (
-                    <a target="__blank" href={value!}>
-                      {value}
-                    </a>
-                  )}
-                </List.Item>
-              )}
-            />
-          ) : undefined
-        }
-        noDataDescrption="Contacts aren't filled in"
-        permissionsSettings={permissionsSettings ? filterPermissions(permissionsSettings) : undefined}
-        isEditingModeEnabled={isEditingModeEnabled}
-        onPermissionsSettingsChange={onPermissionsSettingsChange}
-        detachSettingsOnVisibilityChange
-        profileSettingsContent={
-          <EditForm
-            contacts={contacts}
-            onProfileSettingsChange={onProfileSettingsChange}
-            initialContacts={initialContacts}
-          />
-        }
-      />
-    );
-  }
-}
+  const content = filledContacts.length ? (
+    <List
+      itemLayout="horizontal"
+      dataSource={filledContacts}
+      renderItem={({ name, value, key }: Contact) => (
+        <List.Item>
+          <Text strong>{name}:</Text>{' '}
+          {key !== 'linkedIn' ? (
+            <>
+              {value}
+              {key === 'email' && (!connections.email || !connections.email.enabled) && isEditingModeEnabled ? (
```

Comment:
It seems that you could create enum for `key` values for using as a type and avoiding magic values ;)